### PR TITLE
refactor documentation

### PR DESCRIPTION
**Pull request by NotAShelf**
Introduces a general documentation rework that contains the following
- follow best practices for the nmd documentation generator
- change styling requirements and contain guidelines
- move some contents of CONTRIBUTING.md to the documentation where asciidoc will be used

### CODE REVIEW

**Comment by NotAShelf** [comment]
@horriblename this should be it for the documentation refactor - your PR can be rebased onto this branch and converted to asciidoc if you think the changes are acceptable

**Comment by horriblename** [comment]
LGTM! will rebase my PR soon

File: docs/manual/hacking.adoc [hunk 1]
```diff
@@ -0,0 +1,372 @@
+[[ch-hacking]]
+== Hacking neovim-flake
+
+neovim-flake is designed for developers as much as it is for the end user. I would like any potential contributor to
+be able to propagate their desired changes into the repository without the extra effort. As such, below are guides
+(and guidelines) to streamline  the contribution process and ensure that your valuable input seamlessly integrates
```
Comment (FrothyMarrow) [inline docs/manual/hacking.adoc:6]: Remove the extra space between `streamline  the`

File: docs/manual/hacking.adoc [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+into neovim-flake's development without leaving question marks in your head.
+
+:open-issues: https://github.com/notashelf/neovim-flake/issues
+:new-issue: https://github.com/notashelf/neovim-flake/issues/new
+:seven-rules: https://cbea.ms/git-commit/#seven-rules
+:example-commit-message: https://github.com/nix-community/home-manager/commit/69f8e47e9e74c8d3d060ca22e18246b7f7d988ef
+
+This section is mainly directed towards those who wish to contribute code into neovim-flake. If you wish to instead
+report a bug or discuss a potential feature implementation,first look among the already {open-issues}[open issues] and
```
Comment (FrothyMarrow) [inline docs/manual/hacking.adoc:15]: Add a space after the comma between `implementation,first`

File: docs/manual/hacking.adoc [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+==== Getting started
+
+You naturally would like to start by forking the repository. If you are new to git, have a look at GitHub's
+{fork-a-repo}[Fork a repo guide] for instructions on how you can do this. Once you have a fork of neovim-flake
+you should create a branch starting at the most recent `main` branch.
+Give your branch a reasonably descriptive name, suffixed by its type - i.e `feature/debugger` or `fix/pesky-bug`.
+
+Implement your changes and commit them to the newly created branch and when you are happy with the result and positive
+that it fulfills <<sec-guidelines>>. Once you are confident everything is in order, push the branch to GitHub and
+{create-a-pull-request}[create a pull request], following the template that you will be prompted to fill.
```
Comment (FrothyMarrow) [inline docs/manual/hacking.adoc:31]: Broken link for the `create a pull request`

File: docs/manual/hacking.adoc [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+[[sec-code-style]]
+==== Code Style
+
+**Treewide**
+Keep lines at a reasonable width, ideally 80 characters or less. This also applies to string literals and module
+descriptions and documentation.
+
+**Nix**
+neovim-flake is formatted by the {alejandra}[alejandra] tool and the formatting is checked in the pull
```
Comment (FrothyMarrow) [inline docs/manual/hacking.adoc:132]: Broken link for `alejandra`

File: docs/manual/hacking.adoc [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+:fork-a-repo: https://help.github.com/articles/fork-a-repo/
+:open-issues: https://github.com/notashelf/neovim-flake/issues
+:new-issue: https://github.com/notashelf/neovim-flake/issues/new
+:seven-rules: https://cbea.ms/git-commit/#seven-rules
+:example-commit-message: https://github.com/nix-community/home-manager/commit/69f8e47e9e74c8d3d060ca22e18246b7f7d988ef
+
+This section is mainly directed towards those who wish to contribute code into neovim-flake. If you wish to instead
+report a bug or discuss a potential feature implementation,first look among the already {open-issues}[open issues] and
+If no matching issue exists you may open a {new-issue}[new issue] and describe your problem/request. While creating an
```
Comment (FrothyMarrow) [inline docs/manual/hacking.adoc:17]: Change to lowecase `If`

**Review by FrothyMarrow — COMMENTED** [review]
Fix the broken links and some minor nitpicks, otherwise LGTM 👍

**Review by FrothyMarrow — DISMISSED** [review]

**Review by horriblename — DISMISSED** [review]
LGTM

**Comment by jacekpoz** [comment]
LGTM

**Pull request merged**